CONTRIBUTING.md

```diff
@@ -1,57 +1,169 @@
-- if it is a feature request, please provide a clear argumentation about why you believe this feature
-  should be supported by winit
-
-## Making a pull request
-
-When making a code contribution to winit, before opening your pull request, please make sure that:
-
-- your patch builds with Winit's minimal supported rust version - Rust 1.70.
-- you tested your modifications on all the platforms impacted, or if not possible, detail which platforms
-  were not tested, and what should be tested, so that a maintainer or another contributor can test them
-- you updated any relevant documentation in winit
-- you left comments in your code explaining any part that is not straightforward, so that the
-  maintainers and future contributors don't have to try to guess what your code is supposed to do
-- your PR adds an entry to the changelog file if the introduced change is relevant to winit users.
-
-  You needn't worry about the added entry causing conflicts, the maintainer that merges the PR will
-  handle those for you when merging (see below).
-- if your PR affects the platform compatibility of one or more features or adds another feature, the
-  relevant sections in [`FEATURES.md`](https://github.com/rust-windowing/winit/blob/master/FEATURES.md#features)
-  should be updated.
-
-Once your PR is open, you can ask for a review by a maintainer of your platform. Winit's merging policy
-is that a PR must be approved by at least two maintainers of winit before being merged, including
-at least a maintainer of the platform (a maintainer making a PR themselves counts as approving it).
-
-Once your PR is deemed ready, the merging maintainer will take care of resolving conflicts in
-`CHANGELOG.md` (but you must resolve other conflicts yourself). Doing this requires that you check the
-"give contributors write access to the branch" checkbox when creating the PR.
+# Contribution Guidelines
+
+This document contains guidelines for contributing code to winit. It has to be
+followed in order for your patch to be approved and applied.
+
+## Contributing
+
+Anyone can contribute to winit, however given that it's a cross platform
+windowing toolkit getting certain changes incorporated upstream could be
+challenging.
+
+To save your time it's wise to check already opened [pull requests][prs] and
+[issues][issues]. In general, bug fixes and missing implementations are always
+accepted, however new API proposals should go into the issue first. When in
+doubt contact us on [matrix][matrix] or via opening an issue.
+
+### Submitting your work and handling review
+
+All patches have to be sent on Github as [pull requests][prs]. To simplify your
+life during review it's recommended to check the "give contributors write access
+to the branch" checkbox.
+
+#### Creating commits
+
+When creating a commit, follow these general rules:
+  - Try to limit the first line (title) of the commit message to 50 characters,
```

Comment:
I think that a link here to the Git standards this is taken from would be appreciated.

Reply:
50 characters is not that hard to google, but the source is a bit vague on that. I'd add that multiple editors (nvim, helix) do that sort of indication by default. The 72 chars body is from kernel and in general a lot of inspired by kernel devs, since they know how to write commits. 

https://www.kernel.org/doc/html/latest/process/submitting-patches.html#describe-your-changes

Reply:
Having this linked in this document would be nice